subprojects/model-core/src/test/groovy/org/gradle/internal/reflect/validation/ValidationMessageCheckerTest.groovy

```diff
@@ -0,0 +1,804 @@
+
+Reason: Expected '${location}' to be a directory but it's a file.
+
+Possible solution: Make sure that the 'output' is configured to a directory.
+
+Please refer to https://docs.gradle.org/current/userguide/validation_problems.html#cannot_write_output for more details about this problem.
+"""
+
+        when:
+        render cannotWriteToDir {
+            type('Writer').property('output')
+            dir(location)
+            ancestorIsNotDirectory(ancestor)
+            includeLink()
+        }
+
+        then:
+        outputEquals """
+Type 'Writer' property 'output' is not writable because '${location}' ancestor '${ancestor}' is not a directory.
+
+Reason: Expected '${ancestor}' to be a directory but it's a file.
+
+Possible solution: Make sure that the 'output' is configured to a directory.
+
+Please refer to https://docs.gradle.org/current/userguide/validation_problems.html#cannot_write_output for more details about this problem.
+"""
+    }
+
+    @ValidationTestFor(
+        ValidationProblemId.CANNOT_WRITE_OUTPUT
+    )
+    def "tests output of cannotWriteToFile"() {
+        def location = dummyLocation('/tmp/foo/bar')
+        def ancestor = dummyLocation('/tmp/foo')
+
+        when:
+        render cannotWriteToFile {
+            type('Writer').property('output')
+            file(location)
+            isNotFile()
+            includeLink()
+        }
+
+        then:
+        outputEquals """
+Type 'Writer' property 'output' is not writable because '${location}' is not a file.
+
+Reason: Cannot write a file to a location pointing at a directory.
+
+Possible solution: Configure 'output' to point to a file, not a directory.
+
+Please refer to https://docs.gradle.org/current/userguide/validation_problems.html#cannot_write_output for more details about this problem.
+"""
+
+        when:
+        render cannotWriteToFile {
+            type('Writer').property('output')
+            file(location)
+            ancestorIsNotDirectory(ancestor)
+            includeLink()
+        }
+
+        then:
+        outputEquals """
+Type 'Writer' property 'output' is not writable because '${location}' ancestor '${ancestor}' is not a directory.
+
+Reason: Cannot write a file to a location pointing at a directory.
+
+Possible solution: Configure 'output' to point to a file, not a directory.
+
+Please refer to https://docs.gradle.org/current/userguide/validation_problems.html#cannot_write_output for more details about this problem.
+"""
+    }
+
+    @ValidationTestFor(
+        ValidationProblemId.CANNOT_WRITE_TO_RESERVED_LOCATION
+    )
+    def "tests output of cannotWriteToReservedLocation"() {
+        def reserved = dummyLocation()
+
+        when:
+        render cannotWriteToReservedLocation {
+            type('SomeTransform')
+            forbiddenAt(reserved)
+            includeLink()
+        }
+
+        then:
+        outputEquals """
+Type 'SomeTransform' points to '${reserved}' which is managed by Gradle.
```

Comment:
Hmm..., shouldn't this be `SomeTask`? I also wonder why we don't report the property which points to the reserved location.

Reply:
Only because the test doesn't give a property. I can add one.